nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/data/explorer/PutAzureDataExplorer.java

```diff
@@ -0,0 +1,309 @@
+            throw new ProcessException("Streaming policy is not enabled in database {}" + context.getProperty(DATABASE_NAME).getValue());
+        }
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        String databaseName = context.getProperty(DATABASE_NAME).getValue();
+        String tableName = context.getProperty(TABLE_NAME).getValue();
+        String dataFormat = context.getProperty(DATA_FORMAT).getValue();
+        String mappingName = context.getProperty(MAPPING_NAME).getValue();
+        String routePartiallySuccessfulIngestion = context.getProperty(ROUTE_PARTIALLY_SUCCESSFUL_INGESTION).getValue();
+        String ingestionStatusPollingTimeout = context.getProperty(INGESTION_STATUS_POLLING_TIMEOUT).getValue();
+        String ingestionStatusPollingInterval = context.getProperty(INGESTION_STATUS_POLLING_INTERVAL).getValue();
+        String ignoreFirstRecord = context.getProperty(IGNORE_FIRST_RECORD).getValue();
+        streamingEnabled = context.getProperty(IS_STREAMING_ENABLED).evaluateAttributeExpressions().asBoolean();
+        pollOnIngestionStatus = context.getProperty(POLL_ON_INGESTION_STATUS).evaluateAttributeExpressions().asBoolean();
+
+        boolean isError = false;
+
+        // ingestion into ADX
+        try (final InputStream inputStream = session.read(flowFile)) {
+            StringBuilder ingestLogString = new StringBuilder().append("Ingesting with: ")
+                    .append("dataFormat - ").append(dataFormat).append("|")
+                    .append("ingestionMappingName - ").append(mappingName).append("|")
+                    .append("databaseName - ").append(databaseName).append("|")
+                    .append("tableName - ").append(tableName).append("|")
+                    .append("pollOnIngestionStatus - ").append(pollOnIngestionStatus).append("|")
+                    .append("ingestionStatusPollingTimeout - ").append(ingestionStatusPollingTimeout).append("|")
+                    .append("ingestionStatusPollingInterval - ").append(ingestionStatusPollingInterval).append("|")
+                    .append("routePartiallySuccessfulIngestion - ").append(routePartiallySuccessfulIngestion);
+            getLogger().debug(ingestLogString.toString());
+
+            KustoIngestionResult result = service.ingestData(
+                    new KustoIngestionRequest(streamingEnabled, pollOnIngestionStatus, inputStream, databaseName,
+                            tableName, dataFormat, mappingName, ignoreFirstRecord, ingestionStatusPollingTimeout, ingestionStatusPollingInterval)
+            );
+            getLogger().info("Ingest Status Polling Enabled {} and Ingest Status Polling Enabled {} ", pollOnIngestionStatus, result.toString());
```

Comment:
The `result.toString()` is not necessary. This seems more appropriate as a debug log.
```suggestion
            getLogger().debug("Ingest Status Polling Enabled [{}] and Ingest Status Polling Enabled {}", pollOnIngestionStatus, result);
```